src/main/java/org/quantumbadger/redreader/views/video/ExoPlayerWrapperView.java

```diff
@@ -188,7 +186,66 @@ public ExoPlayerWrapperView(
 							updateProgress();
 						}));
 
-				addButton(playButton.get(), buttons);
+				if (PrefsUtility.pref_behaviour_video_frame_step()) {
+					final AtomicReference<ImageButton> stepBackButton = new AtomicReference<>();
+					final AtomicReference<ImageButton> stepForwardButton = new AtomicReference<>();
+					stepBackButton.set(createButton(
+							context,
+							mControlView,
+							R.drawable.icon_step_back,
+							R.string.video_step_back,
+							view -> {
+								mVideoPlayer.seekTo(mVideoPlayer.getCurrentPosition() - 33);
```

Comment:
You can get frame rate of a video from getVideoFormat and calculate position from it.

Also you probably need to change SeekParameters (https://developer.android.com/reference/androidx/media3/exoplayer/SeekParameters) to NEXT_SYNC/PREVIOUS_SYNC depending on seek direction

Reply:
Thank you for the advice! I changed the calculation to use `getVideoFormat` as requested. I don't think we want to use `SeekParameters` here because that would change the seek behavior in such a way as to make every seek go to a key frame, which could be several seconds before/after where we actually want to seek to in this context. If that's not correct please let me know though.

Reply:
I think you can just reset them back to default value immediately after calling seekTo

Reply:
The issue as I understand it is that the sync points used by SeekParameters aren't necessarily going to be (approximately) the next or previous frame when we're trying to step through a video, they could potentially be hundreds of frames in either direction. Can I ask what your reasoning is for wanting to use SeekParameters here?

Reply:
I thought it would be desirable to seek the next/previous key frame here, but you are right, they could be seconds apart and that's not what we want here